components/core/src/clp/DictionaryWriter.hpp

```diff
@@ -203,35 +202,31 @@ void DictionaryWriter<DictionaryIdType, EntryType>::open_and_preload(
 
     m_max_id = max_id;
 
-    FileReader dictionary_file_reader;
-    FileReader segment_index_file_reader;
+    FileReader dictionary_file_reader{dictionary_path};
+
```

Comment:
Sorry, I didn't spot this earlier. This method was written as a way to reload the dictionary so that we can continue writing an archive, so although we don't use it right now, to maintain correctness, we'd need to also load the segment index, right?

I think an alternative solution is to get rid of `open_and_preload` entirely until we add support for opening and rewriting archives.

Reply:
If I have to choose between the two, maybe removing it.

Reply:
You also need to remove the method prototype.